### PR TITLE
CTCTRADERS-1455 - Reimplement controller endpoint to use customs-reference-data over static values

### PR DESCRIPTION
**Pull request by amaalali**

### CODE REVIEW

File: app/api/controllers/DangerousGoodsCodesController.scala [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      dataRetrieval
+        .getList(UnDangerousGoodsCodeList)
+        .map(
+          _.find(
+            json => (json \ TransportModeListFieldNames.code).as[String] == code
+          )
+        )
+        .map {
+          case Some(data) => Ok(Json.toJson(data))
+          case _ =>
```
Comment (chris-ianson) [inline app/api/controllers/DangerousGoodsCodesController.scala:60]: If we're returning 404 for this should we maybe log an error also (including code)

File: app/api/controllers/PreviousDocumentTypeController.scala [hunk 2]
```diff
@@ -16,35 +16,50 @@
...
+          )
+        )
         .map {
-          documentType =>
-            Ok(Json.toJson(documentType))
-        }
-        .getOrElse {
-          NotFound
+          case Some(data) => Ok(Json.toJson(data))
+          case _ =>
```
Comment (chris-ianson) [inline app/api/controllers/PreviousDocumentTypeController.scala:61]: If we're returning 404 for this should we maybe log an error also (including code)

File: app/api/controllers/SpecialMentionController.scala [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  cc: ControllerComponents,
+  dataRetrieval: DataRetrieval,
+  specialMentionService: SpecialMentionService
+)(implicit ec: ExecutionContext)
+    extends BackendController(cc)
+    with Logging {
+
+  def getAll(): Action[AnyContent] =
+    Action {
+      Ok(Json.toJson(specialMentionService.specialMention))
```
Comment (chris-ianson) [inline app/api/controllers/SpecialMentionController.scala:42]: Do we need a failure case here?

File: app/api/controllers/TransportModeController.scala [hunk 4]
```diff
@@ -21,30 +21,44 @@ import play.api.libs.json.Json
...
+        )
         .map {
-          transportMode =>
-            Ok(Json.toJson(transportMode))
-        }
-        .getOrElse {
-          NotFound
+          case Some(data) => Ok(Json.toJson(data))
+          case _ =>
+            NotFound
```
Comment (chris-ianson) [inline app/api/controllers/TransportModeController.scala:61]: If we're returning 404 for this should we maybe log an error also (including code)

**Review by chris-ianson — CHANGES_REQUESTED** [review]
Couple of comments. What's the reasoning behind a 404 on failure? The reason I'm asking is because none of our consuming services fail gracefully (as far as I'm aware) if they can't get data.I we thinking the calling service will error out anyway (and should kick off a pagerduty)

**Review by chris-ianson — APPROVED** [review]

**Comment by amaalali** [comment]
test this please

**Comment by amaalali** [comment]
test this please

**Pull request merged**